compiler/modules/nback/init.bal

```diff
@@ -0,0 +1,24 @@
+import wso2/nballerina.bir;
+import wso2/nballerina.print.llvm;
+
+public function buildInitModule(bir:ModuleId entryModId, map<bir:FunctionSignature> publicFuncs, llvm:Context llContext) returns llvm:Module|BuildError {
+    llvm:Module llMod = llContext.createModule();
+    llvm:Builder builder = llContext.createBuilder();
+    // SUBSET support cases with no/different entry function
+    string userMainName = "main";
+    check buildMain(entryModId, userMainName, publicFuncs.get(userMainName), llMod, builder);
+    return llMod;
+}
+
+public function buildMain(bir:ModuleId entryModId, string userMainName, bir:FunctionSignature userMainSig, llvm:Module llMod, llvm:Builder builder) returns BuildError? {
+    llvm:FunctionType ty = { returnType: "void", paramTypes: [] };
+    llvm:FunctionDefn func = llMod.addFunctionDefn("_bal_main", ty);
+    builder.positionAtEnd(func.appendBasicBlock());
+
+    string userMainMangledName = mangleInternalSymbol(entryModId, {identifier: userMainName, isPublic: true});
```

Comment:
Mapping constructor missing spaces

Reply:
Ahg fixed, only if we had a style guide.